Pull Request para entrega

ffigueroacasas opened the pull request:
Para que el profesor pueda revisar el codigo y comentarlo

ffigueroacasas committed:
dewsadsa

ffigueroacasas committed:
a ver

P-garrido committed:
sss

P-garrido committed:
ddd

P-garrido committed:
Creación de carpetas

P-garrido committed:
Script BBDD

P-garrido committed:
Capa modelo servicios

P-garrido committed:
Arranque capa controlador

ffigueroacasas committed:
Hice el codigo general para app.js y server.js.
Hice los get, delete y post de users.

P-garrido committed:
Terminé controlador de servicios

P-garrido committed:
Terminé con servicios

P-garrido committed:
empezados algunos cambios en users. cambiar el
insert para que mysql haga el id autoincremental.

ffigueroacasas committed:
desinstalado standard

ffigueroacasas committed:
Merge branch 'main' into pedro

ffigueroacasas committed:
Merge pull request #1 from P-garrido/pedro

Pedro

ffigueroacasas committed:
Terminé el CRUD de usuarios

ffigueroacasas committed:
Termine el CRUD de usuarios

Agustin-Caucino committed:
hice conexion a productos

P-garrido committed:
Termine CRUD servicios-clientes

P-garrido committed:
Merge branch 'pedro' of https://github.com/P-garrido/TP-dsw-back into pedro

P-garrido committed:
Termine CRUD servicios_clientes

P-garrido committed:
Arregle app.js

Agustin-Caucino committed:
arregle coneexion a productos

Agustin-Caucino committed:
agregue updateProduct

P-garrido committed:
Agregué la fecha a clientes_servicios

P-garrido committed:
Merge branch 'main' into pedro

P-garrido committed:
Merge pull request #2 from P-garrido/pedro

Pedro

P-garrido committed:
Ahora el servicio acepta unlos en cant_horaas

P-garrido committed:
Pase servicios a sequelize

P-garrido committed:
Termine de pasar todo a sequelize

P-garrido committed:
Borre servicios y servicios-clientes viejos

P-garrido committed:
Cree archivo para usar sequelize

P-garrido committed:
Merge branch 'main' into pedroSequelize

P-garrido committed:
Merge pull request #5 from P-garrido/pedroSequelize

Pedro sequelize

ffigueroacasas committed:
nada

ffigueroacasas committed:
Merge branch 'main' of https://github.com/P-garrido/TP-dsw-back

ffigueroacasas committed:
Cambiado el model para que use sequelize

P-garrido committed:
Arregle un error en un .json

P-garrido committed:
Agregué mensaje del cliente

P-garrido committed:
Merge pull request #6 from P-garrido/Servicios-msjClli

Agregué mensaje del cliente

P-garrido committed:
Agregue descripcion de servicio

P-garrido committed:
Merge pull request #7 from P-garrido/pedro-jws

Agregue descripcion de servicio

Agustin-Caucino committed:
mi parte del back

Agustin-Caucino committed:
Merge branch 'main' into Agus

Agustin-Caucino committed:
Merge pull request #8 from P-garrido/Agus

mi parte del back

Agustin-Caucino committed:
arregle un par de problemas de funcionamiento

Agustin-Caucino committed:
m

ffigueroacasas committed:
Merge branch 'main' into FranSequalize

ffigueroacasas committed:
Merge pull request #9 from P-garrido/FranSequalize

Cambiado el model de usuarios para que use sequelize

ffigueroacasas committed:
Agregue una llave que faltaba y causaba error

ffigueroacasas committed:
Agregue una llave que faltaba y causaba error

Agustin-Caucino committed:
resolvi problema de claves con sequelize

Agustin-Caucino committed:
Merge branch 'main' into Agus2

Agustin-Caucino committed:
Merge pull request #10 from P-garrido/Agus2

resolvi problema de claves con sequelize

P-garrido committed:
Agregue lo del token

Agustin-Caucino committed:
agregue variable de entorno para la

BD

ffigueroacasas committed:
Agregado el .env a users.js. Borre un par de comas
sobraban en algunos lugares

P-garrido committed:
Puse mis variables de entorno

P-garrido committed:
Termine token

P-garrido committed:
Merge branch 'main' into pedroJWT

P-garrido committed:
Merge pull request #11 from P-garrido/pedroJWT

Pedro jwt

P-garrido committed:
mejore login

Agustin-Caucino committed:
Agregue buscar producto por nombre

larahint committed:
agrego el back de sucursales

larahint committed:
agregue las request

ffigueroacasas committed:
cambios chiquitos

ffigueroacasas committed:
Merge pull request #13 from P-garrido/lara

Lara

P-garrido committed:
Agregue asociaciones user y serv

P-garrido committed:
Merge pull request #14 from P-garrido/pedro-seq

Agregue asociaciones user y serv

P-garrido committed:
agregue onDelete en clientes servicios

Agustin-Caucino committed:
agregue devolucion de cliente en pedidos

ffigueroacasas committed:
Incorporados cambios a la base de datos

ffigueroacasas committed:
Merge branch 'main' of https://github.com/P-garrido/TP-dsw-back

Agustin-Caucino committed:
agregue subir imagenes al serv y obtenerlas

Agustin-Caucino committed:
Merge pull request #16 from P-garrido/images

agregue subir imagenes al serv y obtenerlas

P-garrido committed:
agregado de filtrado servicios

Agustin-Caucino committed:
movi el filtro de productos aca

P-garrido committed:
Merge pull request #17 from P-garrido/pedroEntrega

agregado de filtrado servicios

Agustin-Caucino committed:
Merge pull request #18 from P-garrido/imagenes2

movi el filtro de productos aca

P-garrido committed:
arregle boludeces mias

aotaduy commented on controllers/branches.js:
```diff
@@ -0,0 +1,109 @@
+        return res.status(404).json();
+      }
+
+      // es standard de REST que cuando se elimina un recurso se retorne 204 sin contenido/body
```
no hace falta comentar todo lo que se ahce, se entiende el codigo que sigue podes leer esto 
https://www.baeldung.com/cs/clean-code-comments

aotaduy commented on controllers/branches.js:
```diff
@@ -0,0 +1,109 @@
+      // es standard de REST que cuando se elimina un recurso se retorne 204 sin contenido/body
+      res.status(204).json();
+    }
+    catch (e) {
```
evitemos abreviaciones en el codigo

aotaduy commented on controllers/images.js:
```diff
@@ -0,0 +1,19 @@
+    res.sendFile(imgPath);
+  };
+  uploadImg = async (req, res) => {
+    await pc.uploadImg(req.file.originalname, req.body.id);
```
no me queda claro quie es pc aca

aotaduy commented on controllers/order-products.js:
```diff
@@ -0,0 +1,19 @@
+  createOrderProduct = async (req, res) => {
+    const { id_pedido, id_producto, cantidad } = req.body;
+    const newOrderProductTuple = await this.orderProductsModel.create({
+      id_pedido,
```
en js lo normal es usar camelCase no guiones bajos pero es cuestion de gusto

aotaduy commented on controllers/services-clients.js:
```diff
@@ -0,0 +1,118 @@
+      });
+      res.json(serviceClient);
+    }
+    catch (e) {
```
no deberias devolver algo en este caso al client?

aotaduy commented on controllers/services-clients.js:
```diff
@@ -0,0 +1,118 @@
+      res.json(serviceClient);
+    }
+    catch (e) {
+      console.log(e);
```
evitemos console.log si no son necesarios

aotaduy commented on controllers/services-clients.js:
```diff
@@ -0,0 +1,118 @@
+      res.status(201).json(newServiceClient);
+    }
+    catch (e) {
+      console.log(e);
```
mismo aca de devolver al go al user

aotaduy commented on controllers/services-clients.js:
```diff
@@ -0,0 +1,118 @@
+  }
+
+  delete = async (req, res) => {
+    const idServ = req.params.idServ;
```
si vasd a usar _ te convendria usar la misma convencion para todo podrias evitar estas tres variables

aotaduy commented on controllers/users.js:
```diff
@@ -0,0 +1,114 @@
+    const { id } = req.params;
+    const updatedUser = await this.userModel.update(
+      {
+        nombre_usuario: result.data.nombre_usuario,
```
en mi opinion lo  mejor es tener todo el codigo en ingles para evitar el spanglish que es mas dificil de manejar que el ingles 100%

aotaduy commented on models/order-products.js:
```diff
@@ -0,0 +1,38 @@
+import 'dotenv/config';
+
+const PASSWORD = process.env.PASSWORD;
+const sequelize = new Sequelize('tpdsw', 'root', PASSWORD, {
```
no podrias evitar crear un sequelize para cada modelo ? 

creo que deberias tener solo uno para cada connection o para la app.

aotaduy commented on models/orders.js:
```diff
@@ -0,0 +1,47 @@
+
+const PASSWORD = process.env.PASSWORD;
+
+const sequelize = new Sequelize('tpdsw', 'root', PASSWORD, {
```
no deberias usar root para la aplicacion capaz un usuario sequelize

aotaduy commented on models/orders.js:
```diff
@@ -0,0 +1,47 @@
+      allowNull: false,
+      defaultValue: Sequelize.NOW,
+    },
+    // agregar como clave foranea cuando tenga los demas modelos
```
eliminemos codigo comentado

aotaduy commented on models/services-clients.js:
```diff
@@ -0,0 +1,47 @@
+export const servicesClientsModel = sequelize.define(
+  'services-clients',
+  {
+    id_servicio: {
```
en vez de un id creo que deberias usar una association de sequelize que es la forma mas aporpiada de enlazar dos modelos esto es mas manual

aotaduy commented on models/users.js:
```diff
@@ -0,0 +1,73 @@
+);
+
+userModel.hasMany(servicesClientsModel, { foreignKey: 'id_usuario' });
+servicesClientsModel.belongsTo(userModel, { foreignKey: 'id_usuario' });
```
aca veo que usaste relaciones buenisimo! capaz mi comment anterior esta de mas

aotaduy commented on routes/services-clients.js:
```diff
@@ -0,0 +1,16 @@
+  const servicesClientsRouter = Router();
+  const servicesClientsController = new ServicesClientsController({ servicesClientsModel });
+
+  servicesClientsRouter.get("/", validateToken, servicesClientsController.getAll);
```
no podrias agregar el middleware a nivel router? ya quye todas las rutas lo usan

aotaduy reviewed (changes requested):
Te dejo un feedback

ffigueroacasas committed:
Revisiones
- cree el archivo sequelize.js y exporte desde ahi la variable sequelize
- borre algunos comentarios y console.log()

ffigueroacasas committed:
Eliminacion de console.log
- cambie los console.log(e) en los try...catch para que envie un mensaje de error al cliente

ffigueroacasas committed:
Merge branch 'main' of https://github.com/P-garrido/TP-dsw-back

Agustin-Caucino committed:
agregue endpoint para manejar colecciones

Agustin-Caucino committed:
Merge pull request #19 from P-garrido/imagenes2

agregue endpoint para manejar colecciones

Agustin-Caucino committed:
borre algo innecesario: await pc.uploadImg

Agustin-Caucino committed:
agregue getproductid

Agustin-Caucino committed:
intento resolver schema

Agustin-Caucino committed:
agregue schema

Agustin-Caucino committed:
Merge pull request #20 from P-garrido/ramaAgusC

Rama agus c

Agustin-Caucino committed:
termine lo del token y schema

Agustin-Caucino committed:
corregi schema

Agustin-Caucino committed:
token

P-garrido committed:
borre console log en services controller
